Tests/Unit/HttplugFactoryTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Http\HttplugBundle\Tests\Unit\ClientFactory;
+
+use Http\HttplugBundle\Collector\MessageJournal;
+use Http\HttplugBundle\HttplugFactory;
+use Puli\Discovery\Api\Discovery;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class HttplugFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    public function testEvaluateCondition()
+    {
+        $method = new \ReflectionMethod(HttplugFactory::class, 'evaluateCondition');
```

Comment:
> Generally you should always test the public API. 
I know that is "generally" the idea. Your suggestion would require a lot of mocking and is very likely to break if the class changes. That's why I chose to use reflection to just test the protected function. 

> So what you really want to test in this case is what separate depends parameters return a valid class binding, which of them end up in an exception.

Haha. No :smile:  I really wanted to only test the `evaluateCondition` and leave the rest of the class untested.